src/main/java/io/github/thebusybiscuit/slimefun4/api/events/AsyncMachineProcessCompleteEvent.java

```diff
@@ -20,31 +21,43 @@
 
     private static final HandlerList handlers = new HandlerList();
 
-    private final Block block;
+    private final Location location;
+    private final AContainer container;
     private final MachineRecipe machineRecipe;
 
     @ParametersAreNonnullByDefault
-    public AsyncMachineProcessCompleteEvent(Block block, MachineRecipe machineRecipe) {
+    public AsyncMachineProcessCompleteEvent(Location l, AContainer container, MachineRecipe machineRecipe) {
         super(true);
 
-        this.block = block;
+        this.location = l;
+        this.container = container;
         this.machineRecipe = machineRecipe;
     }
 
     /**
-     * This method returns the {@link Block} of the machine.
+     * This returns the {@link Location} of the machine.
      *
-     * @return the {@link Block} of the machine
+     * @return The {@link Location} of the machine
      */
     @Nonnull
-    public Block getMachine() {
-        return block;
+    public Location getLocation() {
```

Comment:
Why is this changed?

Reply:
Because I thought getting the SlimefunItem instance would be better than just the block. If you look a little lower, there is the new getMachine method.

Reply:
I doubt anyone uses this event yet but in the future avoid changing existent api, at least @Deprecate them and add your new methods, but the old ones should at least work for some time.

Reply:
I know that and agreed. If this wasn't a brand new event, I would've made the old method deprecated. I think I am the only one who uses this rn